pelican/plugins/seo/seo_enhancer/html_enhancer/__init__.py

```diff
@@ -37,14 +37,19 @@ def __init__(self, file, output_path, path, open_graph=False, twitter_cards=Fals
             sitename=_settings.get("SITENAME"),
         )
 
-        # The canonical URL must be built with custom metadata if filled
-        # If not, fallback to the default URL name
+        # The canonical URL must be built with custom metadata if filled.
+        # This can come from either the canonical or save_as field.
+        # If both are absent, fallback to the default URL name
+        canonical = _metadata.get("external_canonical")
         save_as = _metadata.get("save_as")
```

Comment:
This line should now be in the `else` part, we don't need to get it if a canonical is set. It'll avoid a compute for nothing.

Reply:
Done in 14bb5e7f2d1fa67669e797cf35385267839166bc